src/main/java/com/iexec/worker/result/ResultService.java

```diff
@@ -443,4 +427,45 @@ public String computeResultDigest(ComputedFile computedFile) {
         }
         return resultDigest;
     }
+
+    // region Purge
+
+    /**
+     * Purge results from given task, especially its result folder.
+     *
+     * @param chainTaskId ID of the task to purge.
+     * @return {@literal true} if task folder has been deleted
+     * and task data has been cleared from this service;
+     * {@literal false} otherwise.
+     */
+    @Override
+    public boolean purgeTask(String chainTaskId) {
+        final String taskBaseDir = workerConfigService.getTaskBaseDir(chainTaskId);
+
+        resultInfoMap.remove(chainTaskId);
+        FileHelper.deleteFolder(taskBaseDir);
+
+        final boolean deletedInMap = !resultInfoMap.containsKey(chainTaskId);
+        final boolean deletedTaskFolder = !new File(taskBaseDir).exists();
+
+        boolean deleted = deletedInMap && deletedTaskFolder;
+        if (deletedTaskFolder) {
+            log.info("The result of the chainTaskId has been deleted [chainTaskId:{}]", chainTaskId);
+        } else {
+            log.warn("The result of the chainTaskId couldn't be deleted [chainTaskId:{}, deletedInMap:{}, " +
+                            "deletedTaskFolder:{}]",
+                    chainTaskId, deletedInMap, deletedTaskFolder);
+        }
+
+        return deleted;
+    }
+
+    /**
+     * Purge results from all known tasks, especially their result folders.
+     */
+    @Override
+    public void purgeAllTasksData() {
+        resultInfoMap.keySet().forEach(this::purgeTask);
```

Comment:
Are we sure we won't have a problem if we iterate directly over the `keySet` instead of a copy ?

Reply:
We're sure it isn't an issue. You can check the behavior of this method with the following test:
https://github.com/iExecBlockchainComputing/iexec-worker/blob/eba9281658767b34761a605b905c4880fde82917/src/test/java/com/iexec/worker/result/ResultServiceTests.java#L663-L676

Reply:
Could you improve the test by adding several tasks in the `resultInfoMap` and see for yourself what happens ?

Reply:
Oh, right. Thanks for digging! Here's the fix: https://github.com/iExecBlockchainComputing/iexec-worker/pull/476/commits/9846053a82dc4f9b6c8960768ae30a23ede5eafc.

Reply:
Thanks for the fix !